tests/requirements.txt

```
@@ -11,3 +11,4 @@ pytest-pretty==1.2.0
 pytest-timeout==2.1.0
 pytz==2023.3
 numpy==1.25.2; python_version >= "3.9" and python_version < "3.12"
+exceptiongroup==1.1
```

Add python upper bound here perhaps?

```suggestion
exceptiongroup==1.1; python_version < "3.11"
```

---

Added, modified test to not need it in 3.11